### ISSUE TITLE
ulauncher v6 "set hotkey" not working

### ISSUE THREAD

**Issue by choman** [Ulauncher/Ulauncher#1429]
### Environment

ubuntu 24.04 and 24.10, gnome, wayland

### Bug description

I am currently running on both ubuntu 24.04 and 24.10. Trying to move 100% into wayland. Both of these I the gnome desktop environment. 

When trying to set the hotkey that will launch ulauncher v6
- I click the button for "Set hotkey" and the "set new hotkey" popup appears
- I click in the textbox and select my hotkey, ctrl+space.  This shows in the box "Ctrl+Mod2+Space"
- I click on "Save"

These are my observations:
- nothing in the main preferences appears to reflect that I saved a hot key and the "set hotkey" button is still there.
- selecting ctrl+space does not launch uLauncher, with or without the "preferences" widow still active

### Log output

```shell
2024-12-13 10:39:35,778 | INFO | Ulauncher version 6.0.0-beta16 | main.main():53
2024-12-13 10:39:35,778 | INFO | Extension API version 3.0 | main.main():54
2024-12-13 10:39:35,778 | INFO | GTK+ 3.24.41 | main.main():55
2024-12-13 10:39:35,779 | INFO | PyGObject+ 3.48.2 | main.main():56
2024-12-13 10:39:35,779 | INFO | Desktop: UBUNTU:GNOME (WAYLAND) on Ubuntu 24.04.1 LTS | main.main():58
2024-12-13 10:39:35,779 | WARNING | 

╔═════════════════════════════════════════════════════════════════════════════╗
║                  YOU ARE RUNNING A PRE-RELEASE of ULAUNCHER.                ║
║ Please do not report extension API support warnings to extension developers ║
║ We are still in the process of developing and documenting these features    ║
╚═════════════════════════════════════════════════════════════════════════════╝

 | main.main():60
2024-12-13 10:39:35,784 | INFO | Layer shell: No | main.main():71
2024-12-13 10:39:35,784 | INFO | X11 backend: No | main.main():72

```
```


### Communication guidelines

- [X] I have read and agree with Ulauncher's communication guidelines and checked the troubleshooting page for self-help.
- [X] I have tried to understand the error message and help myself, before reporting it.
- [X] My issue description has clear steps for how to reproduce the issue.
- [X] This issue applies to the whole Ulauncher application, not for a specific extension.

### Ulauncher v6

- [X] I have made sure the bug isn't already fixed in v6.

**Comment by friday** [comment]
Looks like Ubuntu changed the DE identifier. If you run `XDG_CURRENT_DESKTOP=GNOME ulauncher` from the CLI it works right?

**Comment by friday** [comment]
Thanks for reporting. Should be fixed in https://github.com/Ulauncher/Ulauncher/releases/tag/v6.0.0-beta17. Deb builds will will be built as part of the release, but Ubuntus servers are slow, so it can take an hour or so.

**Comment by choman** [comment]
Awesome, just installed beta17.  Here are my observations under  ubuntu 24.04 (gnome wayland)
- ctrl_space appears to be the default and working
- Hot Key "area" does not represent the default hotkey
- "Set Hotkey" pops up the dialog with "empty" textbox
- if on e attempts to set "ctrl+space", it launches the main ulauncher app
- multiple occurrences of this causes a crash
- attempted to set hotkey to ctrl+alt+space, lost ability to use hotkey
- removed .config/ulauncher - all back to defaults

**Comment by choman** [comment]
FYI I have a plethora of 26.04 systems, my observations 
- the hot key does not appear to set on any of them 
- some appear to use a default: left ctrl-space
- other like the one that I am on right now: left ctrl-space does not "display" ulauncher


I can get you logs if you  need them

**Comment by friday** [comment]
@choman I'm assuming you're still on Gnome. Open "Keyboard" (settings), scroll to the bottom, press "View and customize Shortcuts", scroll to the bottom again and you should have "Custom Shortcuts". Is "Show Ulauncher" there and what command and shortcut is it bound to?

It should be using `gapplication launch io.ulauncher.Ulauncher` or `ulauncher-toggle` (deprecated) for the command, but `ulauncher show` or `ulauncher toggle` should be working too as of the latest version.

**Comment by choman** [comment]
> [@choman](https://github.com/choman) I'm assuming you're still on Gnome. Open "Keyboard" (settings), scroll to the bottom, press "View and customize Shortcuts", scroll to the bottom again and you should have "Custom Shortcuts". Is "Show Ulauncher" there and what command and shortcut is it bound to?
> 
> It should be using `gapplication launch io.ulauncher.Ulauncher` or `ulauncher-toggle` (deprecated) for the command, but `ulauncher show` or `ulauncher toggle` should be working too as of the latest version.

### epe-sysytem
On a system I will alias as epe-system, it was set to "disabled". Using the keyboard shortcut entry I set it to "ctrl-space". everything worked. I did a test in the ulauncher settings to set the hotkey, 
- it did not show the current setting
- and when I entered ctrl+space, the keyboard shortcut went  back to disabled. 
so it does not appear to be fully using gsetting properly??

#### when set through keyboard shortcuts
I did run the following
```
choman@thor:~$ gsettings get org.gnome.settings-daemon.plugins.media-keys custom-keybindings
['/org/gnome/settings-daemon/plugins/media-keys/custom-keybindings/ulauncher/']
```

Followed by 
```
choman@thor:~$ gsettings get org.gnome.settings-daemon.plugins.media-keys.custom-keybinding:/org/gnome/settings-daemon/plugins/media-keys/custom-keybindings/ulauncher/ binding
'<Mod2>Control_L'
choman@thor:~$ gsettings get org.gnome.settings-daemon.plugins.media-keys.custom-keybinding:/org/gnome/settings-daemon/plugins/media-keys/custom-keybindings/ulauncher/ name
'Show Ulauncher'
choman@thor:~$ gsettings get org.gnome.settings-daemon.plugins.media-keys.custom-keybinding:/org/gnome/settings-daemon/plugins/media-keys/custom-keybindings/ulauncher/ command
'gapplication launch io.ulauncher.Ulauncher'
```

#### When set though ulauncher settings
```
choman@thor:~$ gsettings get org.gnome.settings-daemon.plugins.media-keys.custom-keybinding:/org/gnome/settings-daemon/plugins/media-keys/custom-keybindings/ulauncher/ command
'gapplication launch io.ulauncher.Ulauncher'
choman@thor:~$ gsettings get org.gnome.settings-daemon.plugins.media-keys.custom-keybinding:/org/gnome/settings-daemon/plugins/media-keys/custom-keybindings/ulauncher/ name
'Show Ulauncher'
choman@thor:~$ gsettings get org.gnome.settings-daemon.plugins.media-keys.custom-keybinding:/org/gnome/settings-daemon/plugins/media-keys/custom-keybindings/ulauncher/ binding
'<Primary><Mod2>space'
```
Went set to ```<Primary><Mod2>space``` is when the keyboard shortcuts display "disabled"


Thanks for pointing this out. Hope this helps, will try another system on Monday

**Comment by choman** [comment]
work-system04 
- I applied the same fix and everything is good 
- the gsettings command show the same results

**Comment by friday** [comment]
The problem is perhaps that you have conflicting keybindings. When setting via Ulauncher it can't know that. When setting via the Gnome settings app it might be checking for and disabling the conflicting keybindings.

It might have been introduced by a recent default keybinding in Gnome.

**Comment by choman** [comment]
> The problem is perhaps that you have conflicting keybindings. When setting via Ulauncher it can't know that. When setting via the Gnome settings app it might be checking for and disabling the conflicting keybindings.
> 
> It might have been introduced by a recent default keybinding in Gnome.

I looked at all my keyboard shortcuts and I cannot find a conflicting keybinding.
```
🕙 17:25:31 ❯ gsettings list-recursively org.gnome.desktop.wm.keybindings| grep -i ">space"
org.gnome.desktop.wm.keybindings activate-window-menu ['<Alt>space']
org.gnome.desktop.wm.keybindings switch-input-source ['<Super>space', 'XF86Keyboard']
org.gnome.desktop.wm.keybindings switch-input-source-backward ['<Shift><Super>space', '<Shift>XF86Keyboard']


```

**Comment by friday** [comment]
I believe I figured it out and fixed it ^

`<Mod2>` is num lock. Apparently the way we captured the modifiers was unsafe and also captured num lock and caps lock, but there is a different way to capture them stripping those invalid modifiers, which we use now. I can't test it though. I don't have either of those on my keyboard (I remapped caps to ctrl/esc).

https://github.com/Ulauncher/Ulauncher/releases/tag/v6.0.0-beta33 is building now and should be published within the hour.

Thanks for the patience and debug info!
